client/src/features/sessionsV2/components/SessionButton/ActiveSessionButton.tsx

```diff
@@ -577,12 +556,12 @@ function ModifySessionModalContent({
   useEffect(() => {
     const currentSessionClass = resourcePools
       ?.flatMap((pool) => pool.classes)
-      .find((c) => c.id === currentSessionClassId);
+      .find((c) => `${c.id}` == resource_class_id);
```

Comment:
I think this would be more idiomatic.

```suggestion
      .find((c) => ""+ c.id == resource_class_id);
```

Reply:
This is a javascript "hack" and should not be used. It uses type coercion so this makes code look weird (see things like `"1"+1 == 11`, etc. `${c.id}` is a valid way to convert a `number` to a `string`.